### PR TITLE
[GEOS-10678]: NPE if <settings> element is missing

### PR DESCRIPTION
**Pull request by dieterstueken**
[GEOS-10678](https://osgeo-org.atlassian.net/browse/GEOS-10678)

[GEOS-10678](https://osgeo-org.atlassian.net/browse/GEOS-10678)

When using `data/minimal/global.xml` no `<settings>` element is defined.
This causes a NPE on `XStreamPersister.doUnmarshal()`.
`XStreamPersister.doMarshal()` looks similar, so I also added a precaution null check here.

If the initialization fails, spring destroys its singleton beans again. 
This causes an other NPE on`GeoServerLoader.destroy()` since it was not fully initialized. 
So I added an other NP check here.

# Checklist

- [x]  I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [ ] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by dieterstueken** [comment]
Unit test added, force pushed in front of the fix.
So you can verify the NPE without the fix.

Both other null tests might be obsolete now, but are still preventive (no unit tests for them now).

**Comment by aaime** [comment]
@davidblasby you reported the ticket, could you review the PR?

**Comment by davidblasby** [comment]
ok, looks like all the homepage issues are unrelated to this PR.

File: src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java [hunk 1]
```diff
@@ -2565,7 +2565,7 @@ protected void doMarshal(
                 Object source, HierarchicalStreamWriter writer, MarshallingContext context) {
             GeoServerInfo geoServerInfo = (GeoServerInfo) source;
             SettingsInfo settings = geoServerInfo.getSettings();
-            if (settings.isUseHeadersProxyURL() == null) {
+            if (settings != null && settings.isUseHeadersProxyURL() == null) {
```
Comment (jodygarnett) [inline src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java:2568]: While we can add a null check; the real challenge is that during the load process settings should of been created.

I suspect that this is during the migration of an "old" data directory where only global.xml is defined....

**Review by jodygarnett — CHANGES_REQUESTED** [review]

**Comment by jodygarnett** [comment]
Tested with empty folder and minimal; looks good approving and merging.

**Comment by jodygarnett** [comment]
> With minimal I am still failing to start to to NPE:

Stepping through with debugger GeoServerInfoConverter --> AbstractReflectionConverter.doUnmarshal I noted that GeoServerInfoImpl.toStirng() would also fail due to NPE calling getTitle().

**Review by jodygarnett — APPROVED** [review]
It worked in testing today: release, minimal and empty directory

**Comment by jodygarnett** [comment]
Thanks for the fix

**Pull request merged**